core/src/test/java/greencity/controller/HabitStatisticControllerTest.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+class HabitStatisticControllerTest {
+    private MockMvc mockMvc;
+    @Mock
+    HabitStatisticService habitStatisticService;
+    @InjectMocks
+    HabitStatisticController habitStatisticController;
+
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    long existId = 1;
+
+    @BeforeEach
+    void setUp() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(habitStatisticController)
+                .build();
+    }
+
+    @Test
+    void findAllByHabitId() throws Exception {
+        GetHabitStatisticDto habitStatisticDto = new GetHabitStatisticDto();
+        habitStatisticDto.setHabitStatisticDtoList(Collections.singletonList(new HabitStatisticDto()));
+
+        when(habitStatisticService.findAllStatsByHabitId(1L)).thenReturn(habitStatisticDto);
+        mockMvc.perform(get("/habit/statistic/{habitId}", existId));
+
+        ResponseEntity<GetHabitStatisticDto> response = habitStatisticController.findAllByHabitId(existId);
+
+        Assertions.assertEquals(HttpStatus.OK, response.getStatusCode());
+        Assertions.assertEquals(habitStatisticDto, response.getBody());
+    }
+
+    @Test
+    void findAllStatsByHabitAssignId() throws Exception {
+        List<HabitStatisticDto> habitStatisticDtoList = Collections.singletonList(new HabitStatisticDto());
+
+        when(habitStatisticService.findAllStatsByHabitAssignId(existId)).thenReturn(habitStatisticDtoList);
+        mockMvc.perform(get("/assign/{habitAssignId}", existId));
+
+        ResponseEntity<List<HabitStatisticDto>> response = habitStatisticController.findAllStatsByHabitAssignId(existId);
+
+        Assertions.assertEquals(HttpStatus.OK, response.getStatusCode());
+        Assertions.assertEquals(habitStatisticDtoList, response.getBody());
+    }
+
+    @Test
+    void saveHabitStatistic() throws Exception {
+        AddHabitStatisticDto addHabitStatisticDto = ModelUtils.addHabitStatisticDto();
+        HabitStatisticDto habitStatisticDto = new HabitStatisticDto(
+                existId
+                , addHabitStatisticDto.getHabitRate()
+                , addHabitStatisticDto.getCreateDate()
+                , addHabitStatisticDto.getAmountOfItems()
+                , existId);
+
+        lenient().when(habitStatisticService.saveByHabitIdAndUserId(eq(existId), anyLong(), eq(addHabitStatisticDto)))
+                .thenReturn(habitStatisticDto);
+
+
+        objectMapper.registerModule(new JavaTimeModule());
+        String content = objectMapper.writeValueAsString(addHabitStatisticDto);
+
+        mockMvc.perform(post("/habit/statistic/{habitId}", existId)
```

Comment:
You need to pass .principal() to request to find UserVO using UserArgumentResolver